### ISSUE TITLE
Dynamic width

### ISSUE THREAD

**Issue by timautin** [pengrad/MapScaleView#21]
Hello,

Thank you for this very useful library! I quickly grabbed the code and updated it to make the view's width to be dynamic (ie it is updated when the update(...) method is called). This allows for instance to put a semi-transparent background to the MapScaleView, and to get that background's with to be updated.

Here's are the modifications I made:

Drawer.java:

```
    int getWidth(Scales scales) {
        return Math.round(Math.max(scales.top() == null ? 0 : scales.top().length(), scales.bottom() == null ? 0 : scales.bottom().length()) + strokePaint.getStrokeWidth());
    }
```

MapScaleView.java:

```
    private void updateScales() {
        Scale top, bottom = null;

        if (scaleType == ScaleType.MILES_ONLY) {
            top = mapScaleModel.update(false);
        } else {
            top = mapScaleModel.update(true);
            if (scaleType == ScaleType.BOTH) {
                bottom = mapScaleModel.update(false);
            }
        }

        scales = new Scales(top, bottom);
        requestLayout();
        invalidate();
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int width = measureDimension(desiredWidth(), widthMeasureSpec);
        int height = measureDimension(desiredHeight(), heightMeasureSpec);

        mapScaleModel.setMaxWidth(width);
        drawer.setViewWidth(width);
        updateScales();

        int realWidth = drawer.getWidth(scales);

        setMeasuredDimension(realWidth, height);
    }
```

And here's a possible result (when I move the map, the scale bar's width is updated, including its background):

![Screenshot_1580217035](https://user-images.githubusercontent.com/3846288/73266832-00c6ee80-41d8-11ea-9182-881aba85ef19.png)

Feel free to incorporate them in your code.

**Comment by pengrad** [comment]
Hi, it's definitely good improvement, thanks!

Found 2 issues in your solution - it doesn't respect fixed width (for ex `width=100dp`) and it breaks right-to-left layout.

I've fixed in #22 and will release with new version.

Thanks again, and feel free to open PR next time with any other improvement! :)

**Comment by timautin** [comment]
Great, thanks for release a new version!
Sorry for not posting a PR, I'm a bit in a rush at the moment :)
